### PR TITLE
USHIFT-1521: separate rebase of OCP and LVMS

### PR DESCRIPTION
**Pull request by dhellmann**
Remove the LVMS rebase steps for the overall OCP rebase process and
provide separate, parallel, commands for updating LVMS.

/assign @copejon @pmtk @ggiguash

### CODE REVIEW

**Comment by dhellmann** [comment]
The metal job failed to boot some of the VMs. None of the changes in this PR should have caused that, and we can recheck after reviews.

The pre-submit check for the rebase script passed. [The logs](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift_microshift/2144/pull-ci-openshift-microshift-main-test-rebase/1685736363699212288/artifacts/test-rebase/openshift-microshift-rebase/build-log.txt) show both rebase phases working properly.

**Review by jakobmoellerdev — COMMENTED** [review]
While I understand that these steps will separate the jobs out completely, I feel like this is quite a big clone of basically all rebase.sh steps just to get lvms out of the normal `rebase_to`. 

The existing workaround with a variable inside `rebase.sh` from you did the job quite well and we would only need to remove it from pipelines to get the job running fine without the FIXMEs in https://github.com/openshift/microshift/blob/main/scripts/auto-rebase/rebase.sh#L1162C7-L1171

That way (from what I understand) the process is already purely manual then. Automation could work by simply running the rebase normally and when we do an upstream change in LVMS with a release, propagate by issuing a change in microshift that changes this variable, runs the rebase with that variable (maybe as ENV) changed and then submits a new PR with the existing rebase script.

TLDR: Mainly hesistent that we keep rebase.sh / rebase.md but basically clone every single command/function to separate LVMS out from the rest. If you feel like there is more reasoning to split the two parts, then maybe it would be best if we completely remove it from `rebase.sh` and put it in a `rebase-lvms.sh` so that the domains are truly separate.

File: scripts/auto-rebase/rebase.sh [hunk 1]
```diff
@@ -159,6 +156,8 @@ write_lvms_images_for_arch(){
 }
 
 update_lvms_images(){
+    title "Updating LVMS manifests"
```
Comment (jakobmoellerdev) [inline scripts/auto-rebase/rebase.sh:159]:
```suggestion
    title "Updating LVMS images"
```

File: scripts/auto-rebase/rebase.sh [hunk 2]
```diff
@@ -1141,6 +1145,64 @@ rebase_to() {
...
+    git branch -D "${rebase_branch}" || true
+    git checkout -b "${rebase_branch}"
+
+    update_last_lvms_rebase "${lvms_operator_bundle_manifest}"
+
+    update_lvms_images
+    if [[ -n "$(git status -s pkg/release)" ]]; then
+        title "## Committing changes to pkg/release"
+        git add pkg/release
+        git add packaging/crio.conf.d
```
Comment (jakobmoellerdev) [inline scripts/auto-rebase/rebase.sh:1188]: is this necessary to be updated in lvms rebase (I thought it would only touch assets)? Surprised to see CRIO/pkg/release configuration here

**Comment by dhellmann** [comment]
> While I understand that these steps will separate the jobs out completely, I feel like this is quite a big clone of basically all rebase.sh steps just to get lvms out of the normal `rebase_to`.

Yes, that's exactly what I was going for.

The LVMS and OCP rebase processes are entirely unrelated, except that they update some files in the same directories. I'm trying to untangle them here so they are easier to maintain. The work you did earlier was very helpful in that regard.

> The existing workaround with a variable inside `rebase.sh` from you did the job quite well and we would only need to remove it from pipelines to get the job running fine without the FIXMEs in https://github.com/openshift/microshift/blob/main/scripts/auto-rebase/rebase.sh#L1162C7-L1171
> 
> That way (from what I understand) the process is already purely manual then. Automation could work by simply running the rebase normally and when we do an upstream change in LVMS with a release, propagate by issuing a change in microshift that changes this variable, runs the rebase with that variable (maybe as ENV) changed and then submits a new PR with the existing rebase script.

I did consider that approach. Aside from trying to ease maintenance by separating the code paths, I thought this also made it clearer how to do the LVMS rebase: Run the command, passing the new value as input. 

There's no need for CI or other jobs to keep running the same rebase repeatedly, since there should not be any changes.

> 
> TLDR: Mainly hesistent that we keep rebase.sh / rebase.md but basically clone every single command/function to separate LVMS out from the rest. If you feel like there is more reasoning to split the two parts, then maybe it would be best if we completely remove it from `rebase.sh` and put it in a `rebase-lvms.sh` so that the domains are truly separate.

Eventually I would love to do that. I worried about the CI changes needed to do it all at once in this PR. Maybe we can continue the evolution in that direction next release?

File: scripts/auto-rebase/rebase.sh (hunk 1 above)
Comment (dhellmann) [reply scripts/auto-rebase/rebase.sh:159]: Good catch, I'll fix that in the next revision.

File: scripts/auto-rebase/rebase.sh (hunk 2 above)
Comment (dhellmann) [reply scripts/auto-rebase/rebase.sh:1188]: No, that was a copy-and-paste error from duplicating the commit step. I'll clean that up in the next revision.

**Review by jakobmoellerdev — APPROVED** [review]
Agree with your sentiment, maybe just add one TODO/FIXME so we can go over it and extract it to a separate script next release. Other than that LGTM

**Comment by dhellmann** [comment]
> Agree with your sentiment, maybe just add one TODO/FIXME so we can go over it and extract it to a separate script next release. Other than that LGTM

I opened a jira ticket to remember to do that and included you as a contributor.

File: scripts/auto-rebase/rebase.sh [hunk 3]
```diff
@@ -1161,25 +1223,34 @@ case "$command" in
     to)
         # FIXME: Allow more than 3 arguments until pipelines stop passing LVMS value.
```
Comment (pmtk) [inline scripts/auto-rebase/rebase.sh:1224]: Is this for ART rebase procedure?

Comment (dhellmann) [reply scripts/auto-rebase/rebase.sh:1224]: Yes, the ART pipeline is still passing the extra value.

File: scripts/auto-rebase/rebase.sh (hunk 2 above)
Comment (dhellmann) [reply scripts/auto-rebase/rebase.sh]: Done

File: scripts/auto-rebase/rebase_job_entrypoint.sh [hunk 4]
```diff
@@ -22,23 +22,30 @@ oc registry login --to=/tmp/registry.json
...
+#  correspond to 4.12 LVMS, until the official 4.13 release when both
+#  components will be 4.13.
+release_lvms="v4.13.1-5"
+
+# Since LVMS is not part of the release payload, it is not kept in
+# CI. Use the latest z-stream that coincides with the release
+# payload's X.Y version
+pullspec_release_lvms="registry.access.redhat.com/lvms4/lvms-operator-bundle:${release_lvms}"
+
+./scripts/auto-rebase/rebase.sh lvms-to "${pullspec_release_lvms}"
```
Comment (pmtk) [inline scripts/auto-rebase/rebase_job_entrypoint.sh:51]: Because this is after `rebase.py`, it won't be included in the PR.

Do we run this here, to make sure it continues to work instead of surprising us last minute when we actually need it?

**Comment by pmtk** [comment]
/lgtm
/hold

Just one question

File: scripts/auto-rebase/rebase_job_entrypoint.sh (hunk 4 above)
Comment (dhellmann) [reply scripts/auto-rebase/rebase_job_entrypoint.sh:51]: Yes, that's right. In practice, I expect a person to do the lvms rebase until we get that better automated, so they would submit their own PR.
